Formula/c/clojure-lsp.rb

```diff
@@ -1,10 +1,9 @@
 class ClojureLsp < Formula
   desc "Language Server (LSP) for Clojure"
   homepage "https://github.com/clojure-lsp/clojure-lsp"
-  url "https://github.com/clojure-lsp/clojure-lsp.git",
-      tag:      "2021.09.13-22.25.35",
-      revision: "d564f81e25c71cc370c33d745881af1187f97667"
-  version "20210913T222535"
+  url "https://github.com/clojure-lsp/clojure-lsp/releases/download/2023.10.30-16.25.41/clojure-lsp-standalone.jar"
+  sha256 "4e2fadf51e6b1e64b7b532d6650726f49f438e92f714b34e206268d6503c3370"
```

Comment:
I think our choices are:
1. Use existing JAR for JVM-based usage (above)
2. Move formula to Cask and use GraalVM-built/optimized native executables

We cannot build JAR ourselves easily because upstream switched from Make to Babashka which again requires GraalVM (I tried standalone JAR there but it needs GraalVM libs/classfiles).